packages/react/src/components/Loader/index.test.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import Loader from './';
+import axe from '../../axe';
+
+test('renders a Loader component', () => {
+  render(<Loader data-testid="loader" />);
+  expect(screen.getByTestId('loader')).toBeInTheDocument();
+});
+
+test('handles classNames properly', () => {
+  render(<Loader data-testid="loader" className="baz" />);
+  expect(screen.getByTestId('loader')).toHaveClass('Loader', 'baz');
+});
+
+test('sets aria-hidden if no label is provided', () => {
+  render(<Loader data-testid="loader" />);
+  expect(screen.getByTestId('loader')).toHaveAttribute('aria-hidden', 'true');
+});
+
+test('does not set aria-hidden if a label is provided', () => {
+  render(<Loader data-testid="loader" label="hi" />);
+  expect(screen.getByTestId('loader')).not.toHaveAttribute(
+    'aria-hidden',
+    'false'
+  );
+});
+
+test('sets expected role attributes given an aria-label', () => {
+  render(<Loader data-testid="loader" label="bananas" />);
+  expect(screen.getByTestId('loader')).toHaveAttribute('role', 'alert');
```

Comment:
I think we want to check for the role here:

```suggestion
  expect(screen.getByRole('alert')).toBeInTheDocument()
```

We can also check that the element has an accessible name as well:

```
  expect(screen.getByRole('alert')).toHaveAccessibleName('bananas')
```

Reply:
Hi, the label is within a '<span>' element with the class 'Offscreen'.

`const Loader = React.forwardRef<HTMLDivElement, LoaderProps>(
  ({ className, label, ...props }: LoaderProps, ref) => {
    if (label?.length) {
      props['role'] = 'alert';
      props.children = <Offscreen>{label}</Offscreen>;
    } else {
      props['aria-hidden'] = true;
    }
    return (
      <div ref={ref} className={classNames('Loader', className)} {...props} />
    );
  }
);
`

So, this is what I came up with to find the label. 
`
  expect(screen.getByText('bananas', { selector: '.Offscreen' })).toBeInTheDocument();
`

Please inform me if you are satisfied with this solution, or if you have any alternative suggestions.

Reply:
Ah right, the loader doesn't have an accessible name but wraps offscreen text. In that instance, I think this should be:

```js
expect(screen.getByRole('alert').toHaveTextContent('bananas'))
```